language/move-model/src/ast.rs

```diff
@@ -699,6 +699,20 @@ impl ExpData {
             _ => {}
         });
     }
+
+    pub fn field_usage(&self, usage: &mut BTreeSet<FieldId>) {
```

Comment:
I don't have a comprehensive understanding of the `ExpData` or `Operation` grammars, so would love a second set of eyes on this to make sure I didn't miss a case.

Reply:
I think `Select` and `UpdateField` are the only two cases in the `Operation` enum that touches a field of a struct.

But I don't quite get the `BTreeSet<FieldId>`, maybe what you want is `BTreeSet<(QualifiedId<StructId>, FieldId)>`?

Given the definition of `FieldId`:
```rust
pub struct FieldId(Symbol);
```
it is just a wrapper over `Symbol` and
```
struct S1 {x: u64, y: bool};
struct S2<T> {x: u128, y: T};
```
Will produce the same `FieldId` for `x` and `y` but they refer to quite different things I guess.

Reply:
Good catch! I thought `FieldId`'s were globally unique. I definitely want what you suggest instead.